src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala

```diff
@@ -275,25 +275,33 @@ class Initializer(config: Config) extends SparkSessionWrapper {
     if (overwatchScope.head == "all") config.setOverwatchScope(config.orderedOverwatchScope)
     else config.setOverwatchScope(validateScope(overwatchScope))
 
-    // validate token secret requirements
-    // TODO - Validate if token has access to necessary assets. Warn/Fail if not
-    if (tokenSecret.nonEmpty && !disableValidations && !config.isLocalTesting) {
-      if (tokenSecret.get.scope.isEmpty || tokenSecret.get.key.isEmpty) {
-        throw new BadConfigException(s"Secret AND Key must be provided together or neither of them. " +
-          s"Either supply both or neither.")
-      }
-      val scopeCheck = dbutils.secrets.listScopes().map(_.getName()).toArray.filter(_ == tokenSecret.get.scope)
-      if (scopeCheck.length == 0) throw new BadConfigException(s"Scope ${tokenSecret.get.scope} does not exist " +
-        s"in this workspace. Please provide a scope available and accessible to this account.")
-      val scopeName = scopeCheck.head
-
-      val keyCheck = dbutils.secrets.list(scopeName).toArray.filter(_.key == tokenSecret.get.key)
-      if (keyCheck.length == 0) throw new BadConfigException(s"Key ${tokenSecret.get.key} does not exist " +
-        s"within the provided scope: ${tokenSecret.get.scope}. Please provide a scope and key " +
-        s"available and accessible to this account.")
-
-      config.registerWorkspaceMeta(Some(TokenSecret(scopeName, keyCheck.head.key)))
-    } else config.registerWorkspaceMeta(None)
+
+    rawParams.tokenSecret.map {
+        case databricksSecret: TokenSecret =>
+          val apiToken = SecretTools(databricksSecret).getApiToken
```

Comment:
My understanding that 
```
val apiToken = SecretTools(databricksSecret).getApiToken
```
will throw an exception if scope and key don't exist. So, the below validations don't have sense.
I think we don't need to getApiToken at all or get the validation chain.
Like:
```scala
if (!disableValidations && !config.isLocalTesting) {
            if (databricksSecret.scope.isEmpty || databricksSecret.key.isEmpty) {
              throw new BadConfigException(s"Secret AND Key must be provided together or neither of them. " +
                s"Either supply both or neither.")
            }
            val scopeCheck = dbutils.secrets.listScopes().map(_.getName()).toArray.filter(_ == databricksSecret.scope)
            if (scopeCheck.length == 0) throw new BadConfigException(s"Scope ${databricksSecret.scope} does not exist " +
              s"in this workspace. Please provide a scope available and accessible to this account.")
            val scopeName = scopeCheck.head

            val keyCheck = dbutils.secrets.list(scopeName).toArray.filter(_.key == databricksSecret.key)
            if (keyCheck.length == 0) throw new BadConfigException(s"Key ${databricksSecret.key} does not exist " +
              s"within the provided scope: ${databricksSecret.scope}. Please provide a scope and key " +
              s"available and accessible to this account.")
            val apiToken = SecretTools(databricksSecret).getApiToken
            if (apiToken.length == 0) throw new BadConfigException(s"Value of key ${databricksSecret.key} in scope" +
              s" ${databricksSecret.scope} is empty")
            config.registerWorkspaceMeta(Some(TokenSecret(scopeName, keyCheck.head.key)))
          } else config.registerWorkspaceMeta(None)
```

Reply:
This apiToken derivation indeed redundant and check for None as input for Secrets moved up as described in previous comment